File: by-python/py-boj/5585-거스름돈.py
def solve(paid, mymoney):
    coins = [500, 100, 50, 10, 5, 1]
    cnt = 0
    v = mymoney - paid
    i = 0
    while v > 0:
        if v >= coins[i]:
            v -= coins[i]
            cnt += 1
        else:
            i += 1
    return cnt


if __name__ == '__main__':
    n = int(input())
    print(solve(n, 1000))

File: by-kotlin/kt-boj/solutions/5585-거스름돈.kt
import java.util.*

fun main() = with(Scanner(System.`in`)) {
    val n = nextInt()

    fun solve(myMoney: Int, paid: Int): Int {
        val coins = arrayListOf(500, 100, 50, 10, 5, 1)
        var v = myMoney - paid
        var i = 0
        var cnt = 0
        while (v > 0) {
            if (v >= coins[i]) {
                v -= coins[i]
                cnt += 1
            } else {
                i += 1
            }
        }
        return cnt
    }

    print(solve(1000, n))
}
